lib/assets/javascripts/turbograft/turbolinks.coffee

```diff
@@ -139,7 +139,11 @@ class window.Turbolinks
         if turbohead.hasAssetConflicts()
           return Turbolinks.fullPageNavigate(url.absolute)
         reflectNewUrl url if options.updatePushState
-        turbohead.insertNewAssets(-> updateBody(upstreamDocument, xhr, options))
+        turbohead.waitForAssets().then(->
+          updateBody(upstreamDocument, xhr, options)
+        ).catch(->
+          # A new request appeared during asset download.
```

Comment:
Pushed up an initial implementation of this.  Still trying to make a `turbolinks does not update body when request is superseded` work :/